nupic/research/frameworks/vernon/mixins/oml.py

```diff
@@ -123,21 +155,53 @@ def create_test_test_dataloader(cls, config, dataset):
             pin_memory=torch.cuda.is_available(),
         )
 
+    def pre_task(self, tasks):
+        """Re-initialize task params prior to meta-train training."""
+        super().pre_task(tasks)
+        if not self.reset_task_params:
+            return
+
+        output_params = self.get_named_output_params().values()
+        for p in output_params:
+            if p.dim() == 2:
+                for t in tasks:
+                    task_weights = p[t, :].unsqueeze(0)
+                    nn.init.kaiming_normal_(task_weights)
+
     def post_epoch(self):
         super().post_epoch()
-        if self.should_stop() and self.run_meta_test:
-            # meta-training phase complete, perform meta-testing phase
-            for num_classes_learned in [10, 50, 100, 200, 600]:
-                if num_classes_learned > self.num_classes_eval:
+        if self.current_epoch >= self.epochs - 1 and self.run_meta_test:
+
+            # Accumulate results.
+            table = []
+            headers = ["Num Classes", "Meta-test test", "Meta-test train", "LR"]
+
+            # Meta-training phase complete, perform meta-testing phase
+            for num_classes in self.num_meta_test_classes:
+                if num_classes > self.num_classes_eval:
                     break
 
-                # TODO log results in addition to simply printing them to stdout
-                results = self.run_meta_testing_phase(num_classes_learned)
-                test_train_accs, test_test_accs = results
-                print("Accuracy for meta-testing phase over"
-                      f" {num_classes_learned} num classes.")
-                print("Meta-test training accuracies:", test_train_accs)
-                print("Meta-test tests accuracies:", test_test_accs)
+                # TODO: log results in addition to simply printing them to stdout
+                results = self.run_meta_testing_phase(num_classes)
+                test_train_accs, test_test_accs, lr = results
+
+                print(f"Accuracy for meta-testing phase over {num_classes} num classes")
```

Comment:
We can probably refactor this later to either hide it behind a verbose flag or use a higher level of logging, so we can decide whether or not to print the table.

But I understand now it is the only output, so it is required.